ignite/cmd/plugin.go

```diff
@@ -16,33 +16,46 @@ import (
 	"github.com/ignite/cli/ignite/services/plugin"
 )
 
-// plugins hold the list of plugin declared in the config.
-// A global variable is used so the list is accessible to the plugin commands.
-var plugins []*plugin.Plugin
-
 const (
 	igniteCmdPrefix = "ignite "
 )
 
+// plugins hold the list of plugin declared in the config.
+// A global variable is used so the list is accessible to the plugin commands.
+var plugins []*plugin.Plugin
+
 // LoadPlugins tries to load all the plugins found in configuration.
 // If no configuration found, it returns w/o error.
 func LoadPlugins(ctx context.Context, rootCmd *cobra.Command) error {
-	cfg, err := parseLocalPlugins(rootCmd)
-	if err != nil {
-		// if binary is run where there is no plugins.yml, don't load
-		return nil
+	localCfg, err := parseLocalConfig(rootCmd)
+	// if binary is run where there is no cfg, don't load
+	if err == nil {
+		plugins, err = plugin.Load(ctx, localCfg)
+		if err != nil {
+			return err
+		}
 	}
 
-	// TODO: parse global config
+	globalCfg, err := parseGlobalConfig(rootCmd)
+	// if binary is run where there is no cfg, don't load
+	if err == nil {
+		globalPlugins, err := plugin.Load(ctx, globalCfg)
+		if err != nil {
+			return err
+		}
+		plugins = append(plugins, globalPlugins...)
+	}
 
-	plugins, err = plugin.Load(ctx, cfg)
-	if err != nil {
-		return err
+	if len(plugins) == 0 {
+		return nil
 	}
+
+	// TODO: sort duplicates
```

Comment:
@aljo242 We [discussed](https://github.com/ignite/cli/pull/3181#discussion_r1036843209) about that with @joshLong145, because actually we can even have duplicates without global plugins, if for instance 2 different chains use the same plugin but with different versions.

He suggested we should use the plugin version in the `.ignite/plugins` arborescence, which should sort the duplicates problem, for both situation (global vs local and different chains that use a plugin but with a different version).

That said, this isn't a simple task because the plugin versioning is multiple, it can be the latest commit of default branch (if no version is specified), a tag, a branch and maybe tomorrow a hash (I have a WIP for that). So we need to discuss about the best way to identify a plugin version uniquely, that's why I want to create an issue specifically for that, which means we don't handle that in this PR.

WDYT?

Reply:
To potentially limit complexity could there be a rule for plugins that if a version is specified via path it **must** conform to `semver` such that a package like the following could be used to easily parse and compare version tags:
https://github.com/Masterminds/semver
This would mean that versioning based on commit or branch would bot be supported but would allow for a streamlined implementation and is still quite expressive allowing developers an expressive way of versioning plugins. This would also streamline documentation.

Reply:
Why not, but that means we discard the ability to add a plugin from HEAD, branch or hash right ?

Reply:
I think branches could still be specified, as long as there is no explicit version number within the path. this could be done by not declaring the version within the path to the plugin so 
```
- path: github.com/project/repo
- branchName: brance-name
- version: 0.0.2
```
such that the `URI` can be composed to resolve the plugin source from a series of config properties

Reply:
Currently I like the way we define version with the `@` suffix, because that matches the go get pattern so people are not confused, it just works the same.

If we look at the go mod cache directory (`$GOPATH/pkg/mod`), we can see how they unified the `@tag/branch/hash` into a `@pseudo-version`, that's something I want to investigate in.

Reply:
I like the `@` suffix as well.  

I think this task should go in its own issue.

Reply:
https://github.com/ignite/cli/issues/3242